server/plugin/api.go

```diff
@@ -1224,11 +1226,32 @@ func getRepositoryListByOrg(c context.Context, org string, githubClient *github.
 	return allRepos, http.StatusOK, nil
 }
 
+func getRepository(c context.Context, org string, repo string, githubClient *github.Client) (*github.Repository, int, error) {
+	repository, resp, err := githubClient.Repositories.Get(c, org, repo)
+	if err != nil {
+		return nil, resp.StatusCode, err
+	}
+
+	return repository, http.StatusOK, nil
+}
+
 func (p *Plugin) getRepositories(c *UserContext, w http.ResponseWriter, r *http.Request) {
 	githubClient := p.githubConnectUser(c.Context.Ctx, c.GHInfo)
 
 	org := p.getConfiguration().GitHubOrg
 
+	channelID := r.URL.Query().Get(channelIDParam)
+
+	if channelID == "" {
+		p.writeAPIError(w, &APIErrorResponse{Message: "Bad request: missing channelId", StatusCode: http.StatusBadRequest})
```

Comment:
Add a log statement

Reply:
Also, I dont think we need to stop the repository response in this case. As the repo list is not dependent on the channelID

Reply:
For the current scenario, we don't need to keep the channelID as an optional field, returning from here must not create an issue for us.